state/src/interface/task_queue.rs

```diff
@@ -122,9 +122,25 @@ impl State {
         self.send_proposal(StateTransition::TransitionTask { task_id, state })
     }
 
-    /// Pause a task queue
-    pub fn pause_task_queue(&self, key: &TaskQueueKey) -> Result<ProposalWaiter, StateError> {
-        self.send_proposal(StateTransition::PreemptTaskQueue { key: *key })
+    /// Pause a task queue placing the given task at the front of the queue
+    #[instrument(name = "propose_preempt_task_queue", skip_all, err, fields(task_id, task = %task.display_description()))]
+    pub fn pause_task_queue(
+        &self,
+        key: &TaskQueueKey,
+        task: TaskDescriptor,
+    ) -> Result<ProposalWaiter, StateError> {
+        // Pick a task ID and create a task from the description
+        let id = TaskIdentifier::new_v4();
```

Comment:
hmm, we should have already created a task ID for the preemptive task before we propose pausing the queue. At the very least for a consistent view of traces, could we have that be passed in here and used? Not sure if it breaks some assumptions to use the same task ID on two different queues, though

Reply:
yeah that works, done